ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -758,41 +758,7 @@ let update_allowed_operations ~__context ~self =
     | _ ->
         accu
   in
-  let allowed =
-    List.fold_left check []
-      [
-        `snapshot
-      ; `copy
-      ; `clone
-      ; `revert
-      ; `checkpoint
-      ; `snapshot_with_quiesce
-      ; `start
-      ; `start_on
-      ; `pause
-      ; `unpause
-      ; `clean_shutdown
-      ; `clean_reboot
-      ; `hard_shutdown
-      ; `hard_reboot
-      ; `suspend
-      ; `resume
-      ; `resume_on
-      ; `export
-      ; `destroy
-      ; `provision
-      ; `changing_VCPUs_live
-      ; `pool_migrate
-      ; `migrate_send
-      ; `make_into_template
-      ; `changing_static_range
-      ; `changing_shadow_memory
-      ; `changing_dynamic_range
-      ; `changing_NVRAM
-      ; `create_vtpm
-      ; `metadata_export
-      ]
-  in
+  let allowed = List.fold_left check [] API.vm_operations__all in
```

Comment:
This may now include operations that were previously missed from the hand-written list (like `metadata_export` was until recently). I would be good to do a sanity check to see which operations are "new" and if they are handled correctly.

Reply:
I did add create_vtpm to the check function... now I see this list missing it made it irrelevant when updating allowed_operations :(

You're right, this might not take some operations into consideration